src/test/java/io/github/oliviercailloux/y2018/apartments/utils/JsonConvertTest.java

```diff
@@ -53,17 +55,23 @@ void apartmentToJsonTest() throws IOException, URISyntaxException {
 	/**
 	 * Tests getAddressFromJson function. Verifies if the address extracted by the
 	 * function corresponds to the expected address.
-	 *
-	 * @throws FileNotFoundException if the file doesn't exists.
-	 * @throws IOException           if the file can't be convert into JSON format.
+	 * 
+	 * @throws InvalidObjectException In case the <code>features</code> field is
+	 *                                empty
 	 */
 	@Test
-	void getAddressFromJsonTest() {
-		String adressJson = "{\"data\":{\"latitude\":48.91777636365895,\"longitude\":2.4954686289120067,\"formattedAddress\":\"Allée de Turenne, Nonneville, Les Pavillons-sous-Bois, Le Raincy, Seine-Saint-Denis, Île-de-France, France métropolitaine, 93320, France\",\"country\":\"France\",\"city\":\"Les Pavillons-sous-Bois\",\"state\":\"Île-de-France\",\"zipcode\":\"93320\",\"streetName\":\"Allée de Turenne\",\"countryCode\":\"FR\",\"neighbourhood\":\"\",\"provider\":\"openstreetmap\"},\"address\":\"1 Allée de Turenne, 93320 Les Pavillons-sous-Bois\"}";
-		String adressJsonWrong = "{\"data\":{\"latitude\":48.91777636365895,\"longitude\":2.4954686289120067,\"formattedAddress\":\"Allée de Turenne, Nonneville, Les Pavillons-sous-Bois, Le Raincy, Seine-Saint-Denis, Île-de-France, France métropolitaine, 93320, France\",\"country\":\"France\",\"city\":\"Les Pavillons-sous-Bois\",\"state\":\"Île-de-France\",\"zipcode\":\"93320\",\"streetName\":\"Allée de Turenne\",\"countryCode\":\"FR\",\"neighbourhood\":\"\",\"provider\":\"openstreetmap\"},\"address\":\"\"}";
-
-		assertEquals("1 Allée de Turenne, 93320 Les Pavillons-sous-Bois", JsonConvert.getAddressFromJson(adressJson));
-		assertThrows(IllegalArgumentException.class, () -> JsonConvert.getAddressFromJson(adressJsonWrong));
+	void getAddressFromJsonTest() throws InvalidObjectException {
```

Comment:
Bien que moins académique pour les puristes du test unitaire, j’envisagerais plutôt ici de tester l’appel à l’API d’adresse : envoyez une requête et vérifiez l’attendu. Cela permettra à ce test d’échouer si le format renvoyé par l’API change sans prévenir, par exemple, ce qui le rend plus utile (c’est vraiment ça qu’on veut tester).

Reply:
Add in 5c194b973afdfdd596950971a890493a6c30ba46